CRM/Activity/BAO/Activity.php

```diff
@@ -317,6 +317,8 @@ public static function create(&$params) {
 
     // CRM-9137
     if (!empty($params['id'])) {
+      $activity = CRM_Activity_DAO_Activity::findById($params['id']);
+
       CRM_Utils_Hook::pre('edit', 'Activity', $activity->id, $params);
```

Comment:
@jitendrapurohit You are correct in that would be enough for the pre hook.

We do however pass the Activity object later down the track to the post hook, so I think loading the whole activity object here is the best way to go.

Thoughts?